### PR TITLE
8339906: Open source several AWT focus tests - series 4

### PR DESCRIPTION
**Pull request by prsadhuk**
Opensource few Focus tests

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8339906](https://bugs.openjdk.org/browse/JDK-8339906): Open source several AWT focus tests - series 4 (**Bug** - P4)

### Reviewers
 * [Phil Race](https://openjdk.org/census#prr) (@prrace - **Reviewer**) Review applies to [509ccf65](https://git.openjdk.org/jdk/pull/21051/files/509ccf6553db51df72403be533d4e0bbce529d11)
 * [Abhishek Kumar](https://openjdk.org/census#abhiscxk) (@kumarabhi006 - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/21051/head:pull/21051` \
`$ git checkout pull/21051`

Update a local copy of the PR: \
`$ git checkout pull/21051` \
`$ git pull https://git.openjdk.org/jdk.git pull/21051/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 21051`

View PR using the GUI difftool: \
`$ git pr show -t 21051`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/21051.diff">https://git.openjdk.org/jdk/pull/21051.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/21051#issuecomment-2357442472)

### CODE REVIEW

File: test/jdk/java/awt/Focus/ComponentLostFocusTest.java [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+/*
+ * @test
+ * @bug 4982943
+ * @key headful
+ * @summary focus lost in text fields or text areas, unable to enter characters from keyboar
```
Comment (prrace) [inline test/jdk/java/awt/Focus/ComponentLostFocusTest.java:28]: keyboar -> keyboard

**Review by prrace — APPROVED** [review]

File: test/jdk/java/awt/Focus/ComponentLostFocusTest.java (hunk 1 above)
Comment (prsadhuk) [reply test/jdk/java/awt/Focus/ComponentLostFocusTest.java]: @prrace Fixed but it revokes the approval....Please reapprove or someone "R"eviewer awake in this TZ to approve this minor change..

File: test/jdk/java/awt/Focus/AltTabEventsTest.java [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+           key combination then appropriate window events are generated. Also, when
+           user interacts with Menu bar, Popup menu, Choice then no excessive window
+           event is generated.
+
+           After test started you will see Frame('Test for 4524015')-F1 with some
+           components and Frame('Another frame')-F2 with no components.
+           1. Make F1 active by clicking on it.
+           2. Press Alt-tab.
+           In the messqge dialog area you should see that
+           WINDOW_DEACTIVATED,WINDOW_LOST_FOCUS event were generated.
```
Comment (kumarabhi006) [inline test/jdk/java/awt/Focus/AltTabEventsTest.java:61]:
```suggestion
           WINDOW_DEACTIVATED, WINDOW_LOST_FOCUS event were generated.
```

File: test/jdk/java/awt/Focus/AltTabEventsTest.java [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
+
+           else the test PASSED.""";
+
+    public static void main(String[] args) throws Exception {
+        PassFailJFrame.builder()
+                .title("AltTabEventsTest Instructions")
+                .instructions(INSTRUCTIONS)
+                .rows((int) INSTRUCTIONS.lines().count() + 5)
+                .columns(35)
+                .testUI(AltTabEventsTest::createTestUI)
```
Comment (kumarabhi006) [inline test/jdk/java/awt/Focus/AltTabEventsTest.java:80]: can be changed to `new Test();` and thus leading to remove the `createTestUI()` method.

File: test/jdk/java/awt/Focus/AltTabEventsTest.java [hunk 4]
```diff
@@ -0,0 +1,155 @@
...
+
+    void println(String messageIn) {
+        PassFailJFrame.log(messageIn);
+    }
+
+    public Test() {
+        super("Test for 4524015");
+        WindowAdapter wa = new WindowAdapter() {
+                public void windowActivated(WindowEvent e) {
+                    println(e.toString());
```
Comment (kumarabhi006) [inline test/jdk/java/awt/Focus/AltTabEventsTest.java:104]: can be replaced with `PassFailJFrame.log` everywhere. Anyways there is a repetition of `println`.
```suggestion
                    PassFailJFrame.log(e.toString());
```

File: test/jdk/java/awt/Focus/ComponentLostFocusTest.java [hunk 5]
```diff
@@ -0,0 +1,138 @@
...
+    Frame frame = new Frame("Frame");
+    TextField tf = new TextField("Text Field");
+    static Robot r;
+    Dialog dialog = null;
+    volatile boolean passed;
+
+    public ComponentLostFocusTest() {
+        try{
+            r = new Robot();
+        } catch(Exception e){
```
Comment (kumarabhi006) [inline test/jdk/java/awt/Focus/ComponentLostFocusTest.java:56]:
```suggestion
        } catch (Exception e) {
```

File: test/jdk/java/awt/Focus/ComponentLostFocusTest.java [hunk 6]
```diff
@@ -0,0 +1,138 @@
...
+
+        dialog = new Dialog(frame, "Dialog", true);
+
+        frame.addWindowFocusListener(new WindowAdapter() {
+            public void windowGainedFocus(WindowEvent e) {
+                System.out.println("Frame gained focus: "+e);
+            }
+        });
+
+        frame.setLayout (new FlowLayout ());
```
Comment (kumarabhi006) [inline test/jdk/java/awt/Focus/ComponentLostFocusTest.java:69]:
```suggestion
        frame.setLayout (new FlowLayout());
```

File: test/jdk/java/awt/Focus/ComponentLostFocusTest.java [hunk 7]
```diff
@@ -0,0 +1,138 @@
...
+        frame.add(tf);
+        frame.setSize(400,300);
+        frame.setVisible(true);
+        frame.setLocationRelativeTo(null);
+        frame.validate();
+    }
+
+    public void doTest() {
+        // Do requesting focus to the modal dialog in order to after that
+        // to do requesting focus to the frame
```
Comment (kumarabhi006) [inline test/jdk/java/awt/Focus/ComponentLostFocusTest.java:79]: Doesn't sound correct, can it be rephrased?

File: test/jdk/java/awt/Focus/ComponentLostFocusTest.java [hunk 8]
```diff
@@ -0,0 +1,138 @@
...
+        // Returning the focus to the initial frame will work correctly after the fix
+        System.out.println("disposing.... ");
+        dialog.dispose();
+
+        r.delay(1000);
+
+        // We want to track the GAIN_FOCUS from this time
+        tf.addFocusListener(new FocusAdapter() {
+            public void focusGained(FocusEvent e) {
+                System.out.println("TextField gained focus: "+e);
```
Comment (kumarabhi006) [inline test/jdk/java/awt/Focus/ComponentLostFocusTest.java:103]:
```suggestion
                System.out.println("TextField gained focus: " + e);
```

File: test/jdk/java/awt/Focus/ComponentLostFocusTest.java [hunk 9]
```diff
@@ -0,0 +1,138 @@
...
+public class ComponentLostFocusTest {
+
+    Frame frame = new Frame("Frame");
+    TextField tf = new TextField("Text Field");
+    static Robot r;
+    Dialog dialog = null;
+    volatile boolean passed;
+
+    public ComponentLostFocusTest() {
+        try{
```
Comment (kumarabhi006) [inline test/jdk/java/awt/Focus/ComponentLostFocusTest.java:54]:
```suggestion
        try {
```

File: test/jdk/java/awt/Focus/ComponentLostFocusTest.java [hunk 10]
```diff
@@ -0,0 +1,138 @@
...
+            public void focusGained(FocusEvent e) {
+                System.out.println("TextField gained focus: "+e);
+                passed = true;
+            }
+        });
+
+        doRequestFocusToTextField();
+
+        System.out.println("Focused window: " + KeyboardFocusManager.getCurrentKeyboardFocusManager().getFocusedWindow());
+        System.out.println("Focus owner: " + KeyboardFocusManager.getCurrentKeyboardFocusManager().getFocusOwner());
```
Comment (kumarabhi006) [inline test/jdk/java/awt/Focus/ComponentLostFocusTest.java:111]: please limit it to 80 cols.

File: test/jdk/java/awt/Focus/ComponentLostFocusTest.java [hunk 11]
```diff
@@ -0,0 +1,138 @@
...
+
+        if (!passed) {
+            throw new RuntimeException("TextField got no focus! Test failed.");
+        }
+    }
+
+    private void doRequestFocusToTextField(){
```
Comment (kumarabhi006) [inline test/jdk/java/awt/Focus/ComponentLostFocusTest.java:118]:
```suggestion
    private void doRequestFocusToTextField() {
```

File: test/jdk/java/awt/Focus/ComponentLostFocusTest.java [hunk 12]
```diff
@@ -0,0 +1,138 @@
...
+        // do activation using press title
+        Point loc = frame.getLocationOnScreen();
+        r.mouseMove(loc.x + frame.getWidth()/2, loc.y + frame.getInsets().top/2);
```
Comment (kumarabhi006) [inline test/jdk/java/awt/Focus/ComponentLostFocusTest.java:121]:
```suggestion
        r.mouseMove(loc.x + frame.getWidth() / 2, loc.y + frame.getInsets().top / 2);
```

File: test/jdk/java/awt/Focus/ComponentLostFocusTest.java [hunk 13]
```diff
@@ -0,0 +1,138 @@
...
+        r.waitForIdle();
+        r.mousePress(InputEvent.BUTTON1_DOWN_MASK);
+        r.mouseRelease(InputEvent.BUTTON1_DOWN_MASK);
+        r.waitForIdle();
+
+        // request focus to the text field
+        tf.requestFocus();
+    }
+
+    public static final void main(String args[]){
```
Comment (kumarabhi006) [inline test/jdk/java/awt/Focus/ComponentLostFocusTest.java:131]:
```suggestion
    public static final void main(String args[]) {
```

File: test/jdk/java/awt/Focus/ComponentLostFocusTest.java (hunk 12 above)
Comment (kumarabhi006) [inline test/jdk/java/awt/Focus/ComponentLostFocusTest.java:121]: Should it be accessed on EDT?

File: test/jdk/java/awt/Focus/AltTabEventsTest.java (hunk 2 above)
Comment (prsadhuk) [reply test/jdk/java/awt/Focus/AltTabEventsTest.java]: ok

File: test/jdk/java/awt/Focus/AltTabEventsTest.java (hunk 3 above)
Comment (prsadhuk) [reply test/jdk/java/awt/Focus/AltTabEventsTest.java]: ok

File: test/jdk/java/awt/Focus/AltTabEventsTest.java (hunk 4 above)
Comment (prsadhuk) [reply test/jdk/java/awt/Focus/AltTabEventsTest.java:100]: it will be same..

File: test/jdk/java/awt/Focus/ComponentLostFocusTest.java (hunk 5 above)
Comment (prsadhuk) [reply test/jdk/java/awt/Focus/ComponentLostFocusTest.java]: ok

File: test/jdk/java/awt/Focus/ComponentLostFocusTest.java (hunk 6 above)
Comment (prsadhuk) [reply test/jdk/java/awt/Focus/ComponentLostFocusTest.java]: ok

File: test/jdk/java/awt/Focus/ComponentLostFocusTest.java (hunk 7 above)
Comment (prsadhuk) [reply test/jdk/java/awt/Focus/ComponentLostFocusTest.java]: removed

File: test/jdk/java/awt/Focus/ComponentLostFocusTest.java (hunk 8 above)
Comment (prsadhuk) [reply test/jdk/java/awt/Focus/ComponentLostFocusTest.java]: ok

File: test/jdk/java/awt/Focus/ComponentLostFocusTest.java (hunk 9 above)
Comment (prsadhuk) [reply test/jdk/java/awt/Focus/ComponentLostFocusTest.java]: ok

File: test/jdk/java/awt/Focus/ComponentLostFocusTest.java (hunk 10 above)
Comment (prsadhuk) [reply test/jdk/java/awt/Focus/ComponentLostFocusTest.java]: ok

File: test/jdk/java/awt/Focus/ComponentLostFocusTest.java (hunk 11 above)
Comment (prsadhuk) [reply test/jdk/java/awt/Focus/ComponentLostFocusTest.java]: ok

File: test/jdk/java/awt/Focus/ComponentLostFocusTest.java (hunk 12 above)
Comment (prsadhuk) [reply test/jdk/java/awt/Focus/ComponentLostFocusTest.java]: ok

File: test/jdk/java/awt/Focus/ComponentLostFocusTest.java (hunk 13 above)
Comment (prsadhuk) [reply test/jdk/java/awt/Focus/ComponentLostFocusTest.java]: ok

File: test/jdk/java/awt/Focus/ComponentLostFocusTest.java (hunk 12 above)
Comment (prsadhuk) [reply test/jdk/java/awt/Focus/ComponentLostFocusTest.java]: modified test to put relevant portion under EDT and also added dispose..

File: test/jdk/java/awt/Focus/AltTabEventsTest.java (hunk 3 above)
Comment (kumarabhi006) [reply test/jdk/java/awt/Focus/AltTabEventsTest.java]: Looks like you missed this.

File: test/jdk/java/awt/Focus/AltTabEventsTest.java (hunk 4 above)
Comment (kumarabhi006) [reply test/jdk/java/awt/Focus/AltTabEventsTest.java:100]: `println` method can be removed if `PassFailJFrame.log` is used directly.

File: test/jdk/java/awt/Focus/AltTabEventsTest.java (hunk 3 above)
Comment (kumarabhi006) [reply test/jdk/java/awt/Focus/AltTabEventsTest.java]: Sorry, it was a mistake on my side. Don't know why but updated test was not shown. It is updated now.

File: test/jdk/java/awt/Focus/ComponentLostFocusTest.java [hunk 14]
```diff
@@ -100,25 +95,16 @@ public void run() {
...
-            throw new RuntimeException("TextField got no focus! Test failed.");
-        }
     }
 
-    private void doRequestFocusToTextField(){
+    private static void doRequestFocusToTextField() {
         // do activation using press title
-        Point loc = frame.getLocationOnScreen();
-        r.mouseMove(loc.x + frame.getWidth()/2, loc.y + frame.getInsets().top/2);
+        r.mouseMove(loc.x + frame.getWidth() / 2, loc.y + frame.getInsets().top / 2);
```
Comment (kumarabhi006) [inline test/jdk/java/awt/Focus/ComponentLostFocusTest.java:107]: `frame.getWidth` also should be on EDT ?

File: test/jdk/java/awt/Focus/ComponentLostFocusTest.java [hunk 15]
```diff
@@ -128,11 +114,38 @@ private void doRequestFocusToTextField(){
         tf.requestFocus();
     }
 
-    public static final void main(String args[]){
-        ComponentLostFocusTest test = new ComponentLostFocusTest();
+    public static final void main(String args[]) throws Exception {
+        r = new Robot();
+        r.setAutoDelay(100);
+
+        ComponentLostFocusTest test;
```
Comment (kumarabhi006) [inline test/jdk/java/awt/Focus/ComponentLostFocusTest.java:121]: Unused variable.

File: test/jdk/java/awt/Focus/ComponentLostFocusTest.java [hunk 16]
```diff
@@ -44,39 +45,33 @@
...
-    public ComponentLostFocusTest() {
-        try{
-            r = new Robot();
-        } catch(Exception e){
-            throw new RuntimeException(e);
-        }
-        r.setAutoDelay(100);
+    static Dialog dialog = null;
+    static volatile boolean passed;
+    static Point loc;
```
Comment (kumarabhi006) [inline test/jdk/java/awt/Focus/ComponentLostFocusTest.java:53]: Should it be volatile also?

File: test/jdk/java/awt/Focus/AltTabEventsTest.java (hunk 4 above)
Comment (prsadhuk) [reply test/jdk/java/awt/Focus/AltTabEventsTest.java:100]: Then PassFailJFrame.log will be repeated, so it's same...dont think it's critical to change..

Comment (kumarabhi006) [reply test/jdk/java/awt/Focus/AltTabEventsTest.java:100]: Yeah, not critical.

File: test/jdk/java/awt/Focus/ComponentLostFocusTest.java (hunk 14 above)
Comment (prsadhuk) [reply test/jdk/java/awt/Focus/ComponentLostFocusTest.java]: ok

File: test/jdk/java/awt/Focus/ComponentLostFocusTest.java (hunk 15 above)
Comment (prsadhuk) [reply test/jdk/java/awt/Focus/ComponentLostFocusTest.java]: removed

File: test/jdk/java/awt/Focus/ComponentLostFocusTest.java (hunk 16 above)
Comment (prsadhuk) [reply test/jdk/java/awt/Focus/ComponentLostFocusTest.java]: ok

**Review by kumarabhi006 — APPROVED** [review]
Hope you have run the automated test in CI.

**Pull request closed without merging**